### PR TITLE
(Servicing) Emit font in app bootstrap locale agnostic

### LINKED ISSUE #6144: ApplicationConfiguration Code Generation bug
* .NET Core Version: 
6.0.0

**Problem description:**
When windows system language is set to a language which uses comma as decimal separator the code generator which creates the ApplicationConfiguration class messes up the decimal separator in emSize for the font

For instance when `<ApplicationDefaultFont>Microsoft Sans Serif, 8.25pt</ApplicationDefaultFont>` is set int the csproj file, the following is generated, which obviously does not compile:
`Application.SetDefaultFont(new Font(new FontFamily("Microsoft Sans Serif"), 8,25f, (FontStyle)0, (GraphicsUnit)3));`

The bug is probably here:
https://github.com/dotnet/winforms/blob/c87540e90d424d126306a7f1909f75672ca6004f/src/System.Windows.Forms.Analyzers/src/System/Windows/Forms/ApplicationConfig.FontDescriptor.cs#L35

It obviously should use invariant culture to create that string.

### PR DESCRIPTION
**Pull request by RussKie**
Fixes #6144.

## Proposed changes

- Ensure that the application bootstrap code is generated in locale independent manner.

## Customer Impact

- Customer with locales where decimal separator is not `.` will now have a correct code emitted by the source generated `ApplicationConfiguration` API.

## Regression? 

- No, new feature introduced in .NET 6 RC2

## Risk

- Minimal

## Test methodology 

- manual
- added unit tests

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6171)

### CODE REVIEW

File: src/System.Windows.Forms.Analyzers.CSharp/src/System/Windows/Forms/Generators/ProjectFileReader.FontConverter.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ internal static class FontConverter
                 {
                     // Get the style index (if any). The size is a bit problematic because it can be formatted differently
                     // depending on the culture, we'll parse it last.
-                    int styleIndex = CultureInfo.InvariantCulture.CompareInfo.IndexOf(font, StylePrefix, CompareOptions.IgnoreCase);
+                    int styleIndex = s_culture.CompareInfo.IndexOf(font, StylePrefix, CompareOptions.IgnoreCase);
```
Comment (danmoseley) [inline src/System.Windows.Forms.Analyzers.CSharp/src/System/Windows/Forms/Generators/ProjectFileReader.FontConverter.cs:53]: curious why (existing) code uses `s_culture`. Why not just CultureInfo.InvariantCulture? AFAIK we do not cache it like this anywhere in the core libraries, and it's easier to read.

Comment (RussKie) [reply src/System.Windows.Forms.Analyzers.CSharp/src/System/Windows/Forms/Generators/ProjectFileReader.FontConverter.cs:53]: Because it is :)
image

Comment (RussKie) [reply src/System.Windows.Forms.Analyzers.CSharp/src/System/Windows/Forms/Generators/ProjectFileReader.FontConverter.cs:53]: In the runtime implementation (where this code is copied from) the culture can be different, so there's no caching. Here we only use the invariant. 
I guess we could have gone with using the invariant everywhere but it makes it a little more verbose (and potentially lead to few extra instructions).

**Comment by RussKie** [comment]
Approved over the email

**Comment by RussKie** [comment]
The fix was applied to 6.0.1, but due to https://github.com/dotnet/windowsdesktop/pull/2362 it hasn't made its way into the 6.0.1 servicing release. This will be shipped in 6.0.2 release.

**Pull request merged**